### PR TITLE
fix: Make Int64.fromDecimalString() act more like parseInt()

### PR DESCRIPTION
**Pull request by bradley-marker-ck**
Make `Int64.fromDecimalString()` behave more like `parseInt()`.

## Description
After releasing v0.16.0, we found that some code was passing in a non-numeric strings and expecting a zero value.

The `parseInt()` function returns `NaN` for non-numeric strings but this was getting converted to `0` in earlier versions that were not checking the input for a valid value. This become an error with v0.16.0. Make it zero again.

Handle strings with a numeric prefix as parseInt() as well. Just convert the numeric prefix, ignoring the rest of the string. The old code would try to convert the whole string to a number without extracting just the numeric parts. This could result in `NaN` high or low parts with unexpected results.

## Motivation and Context
This caused some regressions in early adopters. These were mostly tests for edge cases but delayed releases.

Avoid more widespread regressions.

## How Has This Been Tested?
More extensive unit tests for the edge cases around unexpected strings for `Int64.fromDecimalString()`.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/thrift-server-core/src/main/Int64.ts [hunk 1]
```diff
@@ -258,32 +261,45 @@ export class Int64 implements IInt64 {
...
+            return new Int64(0)
+        }
+        if (Number.isSafeInteger(numVal)) {
+            // Use the number.
+            return new Int64(numVal)
+        }
+
+        // Extract just the number part.
+        const matches = text.match(DEC_REGEX)
+        const numericPart = (matches && matches[1]) || ''
```
Comment (bradley-marker-ck) [inline packages/thrift-server-core/src/main/Int64.ts]: I was trying to avoid the regex for performance reasons.

I could leave just the short number change above to check for NaN and leave out the following changes without the regex if performance trumps some of the odd cases that start with a number but are not all number.

File: packages/thrift-server-core/src/main/Int64.ts [hunk 2]
```diff
@@ -258,32 +261,45 @@ export class Int64 implements IInt64 {
     public static fromDecimalString(text: string): Int64 {
         const negative: boolean = text.charAt(0) === '-'
 
-        if (text.length < (negative ? 17 : 16)) {
-            // The magnitude is smaller than 2^53.
-            return new Int64(+text)
-        } else if (text.length > (negative ? 20 : 19)) {
+        // Short numbers are fast.
+        const numVal = parseInt(text, 10)
```
Comment (bradley-marker-ck) [inline packages/thrift-server-core/src/main/Int64.ts]: I could leave this as just `+text` or `Number(text)`. The difference being that the whole string needs to be a number. However, the hex string handling already behaves more like `parseInt()`. Maybe that was a bad choice and it should have strictly required that the whole string be a hex value. I didn't realize the number conversion and `parseInt()` behaved that differently.
